src/main/java/org/sonar/scm/git/blame/GraphNodeFactory.java

```diff
@@ -43,12 +50,17 @@ public StatefulCommitFactory(@Nullable Set<String> filePathsToBlame) {
    *
    * @return a {link StatefulCommit} for the given commit and the files found.
    */
-  public StatefulCommit create(TreeWalk treeWalk, RevCommit commit) throws IOException {
+  public CommitGraphNode createForCommit(TreeWalk treeWalk, @CheckForNull RevCommit commit) throws IOException {
     MutableObjectId idBuf = new MutableObjectId();
     List<FileCandidate> files = new ArrayList<>();
 
     treeWalk.setRecursive(true);
-    treeWalk.reset(commit.getTree());
+    if (commit != null) {
+      treeWalk.reset(commit.getTree());
+    } else {
+      treeWalk.reset();
```

Comment:
I don't get how we can fall into that scenario. Shouldn't we fail safe if startCommit is not found in Git? otherwise we end up with a CommitGraphNode that has a null commit, which is not correct I think

Reply:
yes, good catch. This was left over from my first attempt to implement it.